### PR TITLE
Remove deprecated `validate_*`

### PR DESCRIPTION
**Pull request by ralfbosz**
This is the same as #65 but now
using datatypes instead of the
validate_legacy function.

### CODE REVIEW

**Comment by ralfbosz** [comment]
> you could also update `has_key` function in [config.pp](https://github.com/voxpupuli/puppet-winlogbeat/blob/22d7fa693d8548be52795132e622452af44a04db/manifests/config.pp#L28) file @ralfbosz

Missed that one, fixed it.

File: manifests/init.pp [hunk 1]
```diff
@@ -25,39 +25,37 @@
...
-  $conf_template        = undef,
-  $download_url         = undef,
-  $install_dir          = $winlogbeat::params::install_dir,
-  $tmp_dir              = $winlogbeat::params::tmp_dir,
+  Optional[String]          $major_version        = undef,
+  String                    $package_ensure       = $winlogbeat::params::package_ensure,
+  String                    $service_ensure       = $winlogbeat::params::service_ensure,
+  Boolean                   $service_enable       = $winlogbeat::params::service_enable,
+  Optional[String]          $service_provider     = $winlogbeat::params::service_provider,
+  String                    $registry_file        = $winlogbeat::params::registry_file,
```
Comment (smortex) [inline manifests/init.pp:35]: Is a `Stdlib::Windowspath` more suitable?

File: manifests/init.pp [hunk 2]
```diff
@@ -25,39 +25,37 @@
...
+  Stdlib::Windowspath       $config_file          = $winlogbeat::params::config_file,
+  Hash                      $outputs              = $winlogbeat::params::outputs,
+  Hash                      $shipper              = $winlogbeat::params::shipper,
+  Hash                      $logging              = $winlogbeat::params::logging,
+  Hash                      $run_options          = $winlogbeat::params::run_options,
+  Optional[String]          $conf_template        = undef,
+  Optional[Stdlib::HTTPUrl] $download_url         = undef,
+  String                    $install_dir          = $winlogbeat::params::install_dir,
+  String                    $tmp_dir              = $winlogbeat::params::tmp_dir,
```
Comment (smortex) [inline manifests/init.pp:44]: Is a Stdlib::Windowspath more suitable?

File: manifests/init.pp [hunk 3]
```diff
@@ -25,39 +25,37 @@
...
-  $beat_name            = $winlogbeat::params::beat_name,
-  $tags                 = $winlogbeat::params::tags,
-  $queue_size           = $winlogbeat::params::queue_size,
-  $max_procs            = $winlogbeat::params::max_procs,
-  $fields               = $winlogbeat::params::fields,
-  $fields_under_root    = $winlogbeat::params::fields_under_root,
-  $metrics              = undef,
+  Boolean                   $use_generic_template = $winlogbeat::params::use_generic_template,
+  Stdlib::Fqdn              $beat_name            = $winlogbeat::params::beat_name,
+  Array                     $tags                 = $winlogbeat::params::tags,
```
Comment (smortex) [inline manifests/init.pp:48]: It's always good if we can qualify this more precisely, e.g. `Array[String[1]]` for an array of non-emty strings.

File: manifests/init.pp [hunk 4]
```diff
@@ -25,39 +25,37 @@
...
+  Integer                   $queue_size           = $winlogbeat::params::queue_size,
```
Comment (smortex) [inline manifests/init.pp:49]: I guess a negative value does not make sense?  Just like with a String, we can set a lower bound for an Integer, e.g.

```suggestion
  Integer[1]                $queue_size           = $winlogbeat::params::queue_size,
```

**Review by smortex — COMMENTED** [review]
Looks good, I detailed some minor issues with the data types.  I have no experience with this module and maybe adding more validation to the various `Hash` would be great, but it is a good start, thanks for this effort!

**Comment by zilchms** [comment]
superseded by #68 
Hey @ralfbosz, thanks for your contribution. I decided to merge your PR and #67 into one PR and fix the changerequests there altogether. I needed these changes for #62 . Thanks for your contribution nonetheless!

**Pull request closed without merging**